appeals/web/src/server/appeals/appeal-details/hearing/add-estimates/__tests__/add-estimates.test.js

```diff
@@ -0,0 +1,197 @@
+ * @param {string} fieldName - The name of the field being tested
+ * @param {string} label - The label used in error messages
+ * @param {Object} validValues - The valid values for other fields
+ * @returns {() => void} A function that runs the validation test cases
+ */
+const behavesLikeADayOrHalfDayField = (fieldName, label, validValues) => {
+	return () => {
+		it('should return 400 when field is empty', async () => {
+			const response = await request
+				.post(`${baseUrl}/2/hearing/add-estimates/estimates`)
+				.send({ ...validValues, [fieldName]: '' });
+
+			expect(response.statusCode).toBe(400);
+			expect(response.text).toContain(`Enter estimated ${label}`);
+		});
+
+		it('should return 400 when field is not numeric', async () => {
+			const response = await request
+				.post(`${baseUrl}/2/hearing/add-estimates/estimates`)
+				.send({ ...validValues, [fieldName]: 'abc' });
+
+			expect(response.statusCode).toBe(400);
+			expect(response.text).toContain(`Estimated ${label} must be a number`);
+		});
+
+		it('should return 400 when field is negative', async () => {
+			const response = await request
+				.post(`${baseUrl}/2/hearing/add-estimates/estimates`)
+				.send({ ...validValues, [fieldName]: '-1' });
+
+			expect(response.statusCode).toBe(400);
+			expect(response.text).toContain(`Estimated ${label} must be between 0 and 99`);
+		});
+
+		it('should return 400 when field is greater than 99', async () => {
+			const response = await request
+				.post(`${baseUrl}/2/hearing/add-estimates/estimates`)
+				.send({ ...validValues, [fieldName]: '100' });
+
+			expect(response.statusCode).toBe(400);
+			expect(response.text).toContain(`Estimated ${label} must be between 0 and 99`);
+		});
+
+		it('should return 400 when field is not in increments of 0.5', async () => {
+			const response = await request
+				.post(`${baseUrl}/2/hearing/add-estimates/estimates`)
+				.send({ ...validValues, [fieldName]: '1.2' });
+
+			expect(response.statusCode).toBe(400);
+			expect(response.text).toContain(`Estimated ${label} must be in increments of 0.5`);
+		});
+
+		it('should accept field in increments of 0.5', async () => {
+			const response = await request
+				.post(`${baseUrl}/2/hearing/add-estimates/estimates`)
+				.send({ ...validValues, [fieldName]: '1.5' });
+
+			expect(response.statusCode).toBe(302);
+		});
+	};
+};
+
+describe('add estimates', () => {
+	beforeEach(installMockApi);
+	afterEach(teardown);
+
+	describe('GET /hearing/add-estimates', () => {
+		// eslint-disable-next-line jest/expect-expect
+		it('should redirect to /hearing/add-estimates/estimates', () => {
+			return new Promise((resolve) => {
+				request
+					.get(`${baseUrl}/2/hearing/add-estimates`)
+					.expect(302)
+					.expect('Location', `${baseUrl}/2/hearing/add-estimates/estimates`)
+					.end(resolve);
+			});
+		});
+	});
+
+	describe('GET /hearing/add-estimates/estimates', () => {
+		const appealId = 2;
+
+		/** @type {*} */
```

Comment:
We're ignoring TS entirely for most of the test files, you may want to do that here too, to avoid needing these annotations (just a thought)